src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/MySqlEngine.java

```diff
@@ -861,14 +863,25 @@ protected void setSchema(final String schema) throws DatabaseEngineException {
     }
 
     @Override
-    public synchronized ResultIterator iterator(String query) throws DatabaseEngineException {
+    public ResultIterator iterator(String query) throws DatabaseEngineException {
+        return iterator(query, properties.getFetchSize(), properties.getSelectQueryTimeout());
+    }
+
+    @Override
+    public ResultIterator iterator(String query, int fetchSize, int readTimeout) throws DatabaseEngineException {
         try {
             getConnection();
             final Statement stmt = conn.createStatement(TYPE_FORWARD_ONLY, CONCUR_READ_ONLY);
-            stmt.setFetchSize(properties.getFetchSize());
+            stmt.setFetchSize(fetchSize);
+            if (readTimeout != NO_TIMEOUT) {
+                stmt.setQueryTimeout(readTimeout);
+            }
```

Comment:
I think that with the changes in #127 (replace DEFAULT_FETCH_SIZE in iterator methods with value from config) and with your changes, this method can be exactly the same as the super method if instead you override `createSelectStatement` to use `conn.createStatement(TYPE_FORWARD_ONLY, CONCUR_READ_ONLY);`.

However, in the MySQL driver those values are the already default (they were default also when this change was introduced), so I think we can simply use `conn.createStatement()`, which makes both `iterator` methods overrides unnecessary at all in MySqlEngine!

Note: this override was introduced in a commit to "Fixes MySqlEngine not being able to iterate using fetch size.", but from my experiments because of CockroachDB, MySQL still doesn't respect the fetch size setting; it seems the only way to make it work is to use server-side cursors, enabled with a flag in the JDBC URL.